odoo/models.py

```diff
@@ -4240,6 +4244,17 @@ def _load_records(self, data_list, update=False):
                 if data.get('xml_id') and not data['xml_id'].startswith(prefix):
                     _logger.warning("Creating record %s in module %s.", data['xml_id'], module)
 
+        if self.env.context.get('import_file'):
+            existing_modules = self.env['ir.module.module'].search([]).mapped('name')
+            for data in to_create:
+                xml_id = data.get('xml_id')
+                if xml_id:
+                    module_name = xml_id.split('.')[0]
+                    if module_name in existing_modules:
```

Comment:
This is a good point, but it seems the same behavior mentioned in https://github.com/odoo/odoo/pull/130825#discussion_r1284195235 applies: if the xml id is unprefixed and something like "test", it will be `__import__.test` at this point. I don't mind making the code more robust by using the partition call though, it's more robust and generic than my original.

Reply:
I think the `partition` approach is more robust against refactoring (where maybe the fact that there's currently an `__import__` prefix added if there is none might not be true anymore at some point in the future for this logic). In that case the UserError can't be raised due to having no prefix, which is fine.

By adding the assertion requiring the prefix, an exception would occur in that case, and doing that kind of validation for this function as a whole isn't the responsibility of this added bit code I think.